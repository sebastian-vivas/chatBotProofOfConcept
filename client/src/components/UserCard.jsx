function UserCard(props) {
    const userpic = {
        width: "45px", // Adjust width as needed
        height: "45px", // Adjust height as needed
        borderRadius: "50%", // To make the image round
        boxShadow: "0 6px 4px rgba(0, 0, 0, 0.1)" // Add a subtle shadow
      };
    
      const cardStyle = {
    
       
    
        padding: "10px 11px 10px 16px",
        borderRadius: "0px 0px 10px 0px",
        boxShadow: "2px 3px 5px black",
        textAlign: "center",
      
       
        display: "flex",
        width: "100%"
    };

    
      const headingStyle = {
        
        fontSize: "px",
        margin: "5%, 10%, 5%, 5%",
        padding: "5%"
        
      };
    
  
    return (
      <div style={cardStyle}>
        <img  style={userpic} src={props.imgUser}></img>
        <h1 style={headingStyle}>Welcome, {props.name}! </h1>
      </div>
    );
  }
  
  export default UserCard;