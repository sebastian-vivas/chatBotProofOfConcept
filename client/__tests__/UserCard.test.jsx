import React from 'react';

import UserCard from '../src/components/UserCard.jsx';
import { test, expect } from 'vitest'; 



test('renders user card with name and image', () => {


    const user = { //creating a user name with and img to be render on DOM when rending app
        name: "Daniel Garcia",
        img: "https://i.pravatar.cc/150?u=a04258114e29026702d",
    };

    const result = <UserCard name={user.name} img={user.img} />; // making varible with component and props 


    expect(result).toMatchSnapshot(); // we expect that component to match out snapshot be the same
});