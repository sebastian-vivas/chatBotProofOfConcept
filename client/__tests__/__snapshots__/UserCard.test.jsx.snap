// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders user card with name and image 1`] = `
<UserCard
  img="https://i.pravatar.cc/150?u=a04258114e29026702d"
  name="Daniel Garcia"
/>
`;
